centreon/tests/e2e/cypress/e2e/Resources-status/02-acknowledgments/index.ts

```diff
@@ -0,0 +1,649 @@
+      status: initial_status
+    };
+
+    submitCustomResultsViaClapi(hostStatus);
+
+    checkThatFixtureHostsExistInDatabase(
+      hostChildInAcknowledgementName,
+      `submit_${hostChildInAcknowledgementName}_${initial_status}`,
+      [hostStatus]
+    );
+
+    cy.refreshListing();
+
+    cy.contains(hostChildInAcknowledgementName)
+      .parent()
+      .parent()
+      .find('input[type="checkbox"]:first')
+      .click();
+  }
+);
+
+Given(
+  'a resource of service is selected with {string}',
+  (initial_status: string) => {
+    const serviceStatus = {
+      host: hostInAcknowledgementName,
+      output: `submit_${serviceInAcknowledgementName}_${initial_status}`,
+      service: serviceInAcknowledgementName,
+      status: initial_status
+    };
+
+    submitCustomResultsViaClapi(serviceStatus);
+
+    checkThatFixtureServicesExistInDatabase(
+      serviceInAcknowledgementName,
+      `submit_${serviceInAcknowledgementName}_${initial_status}`,
+      [serviceStatus]
+    );
+
+    cy.refreshListing();
+
+    cy.contains(serviceInAcknowledgementName)
+      .parent()
+      .parent()
+      .find('input[type="checkbox"]:first')
+      .click();
+  }
+);
+
+When('"sticky" checkbox is {string} in the form', (check: string) => {
+  switch (check) {
+    case 'unchecked':
+      cy.getByLabel({ label: 'Sticky' }).uncheck();
+      cy.getByLabel({ label: 'Sticky' }).should('not.be.checked');
+      break;
+    default:
+      cy.getByLabel({ label: 'Sticky' }).check();
+      cy.getByLabel({ label: 'Sticky' }).should('be.checked');
+      break;
+  }
+});
+
+When('"persistent" checkbox is {string} in the form', (check: string) => {
+  switch (check) {
+    case 'unchecked':
+      cy.getByLabel({ label: 'Persistent' }).uncheck();
+
+      cy.getByLabel({ label: 'Persistent' }).should('not.be.checked');
+      break;
+    default:
+      cy.getByLabel({ label: 'Persistent' }).check();
+
+      cy.getByLabel({ label: 'Persistent' }).should('be.checked');
+      break;
+  }
+});
+
+When('the {string} resource is marked as acknowledged', (resource: string) => {
+  let resourceName = '';
+
+  switch (resource) {
+    case 'service':
+      resourceName = serviceInAcknowledgementName;
+      break;
+    default:
+      resourceName = hostChildInAcknowledgementName;
+      break;
+  }
+
+  cy.getByLabel({ label: 'State filter' }).click();
+
+  cy.get('[data-value="all"]').click();
+
+  cy.waitUntil(
+    () => {
+      return cy
+        .refreshListing()
+        .then(() => cy.contains(resourceName))
+        .parent()
+        .then((val) => {
+          return (
+            val.css('background-color') === actionBackgroundColors.acknowledge
+          );
+        });
+    },
+    {
+      timeout: 15000
+    }
+  );
+});
+
+When(
+  'the {string} status changes to {string}',
+  (resource: string, changed_status: string) => {
+    clearCentengineLogs();
+    let status;
+
+    switch (resource) {
+      case 'service':
```

Comment:
Instead of using switch statement, we use if statement. Is it possible to do this?